src/ast/mod.rs

```diff
@@ -477,6 +477,15 @@ pub enum CastKind {
     DoubleColon,
 }
 
+/// `EXTRACT` syntax types.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum ExtractSyntax {
+    From,
+    Comma
```

Comment:
Can we add the example syntax showing the difference between the two? (Also for the comma, lets add a link to the snowflake docs that mention the syntax)